source/agora/node/Ledger.d

```diff
@@ -277,44 +277,54 @@ public class Ledger
         if (!this.node_config.is_validator)
             return;
 
-        // if we received another transaction while we're nominating, don't nominate again.
-        // todo: when we change nomination to be time-based (rather than input-based),
-        // then remove this part as it will be handled by a timer
-        if (this.is_nominating)
-            return;
+        this.is_nominating = true;
+        scope (exit) this.is_nominating = false;
 
-        if (this.pool.length < Block.TxsInBlock)
+        Set!Transaction txs;
+        if (!this.prepareNominatingSet(txs))
             return;
 
-        Hash[] hashes;
-        Set!Transaction txs;
-        ulong expect_height = this.getBlockHeight() + 1;
+        // note: we are not passing the previous tx set as we don't really
+        // need it at this point (might later be necessary for chain upgrades)
+        auto slot_idx = this.last_block.header.height + 1;
+        this.nominate(slot_idx, Set!Transaction.init, txs);
```

Comment:
Looks like we might want to change the signature of `NominateDg` at some point if it's always called with `.init` as second arg ?

Reply:
This delegate is removed in a further commit. I originally had it part of one commit, but the changes were very hard to review, so I split it up.